docs/tutorial/dark-mode.md

```diff
@@ -80,65 +80,74 @@ Add the following lines to the `index.html` file:
 </html>
 ```
 
-Next, add [event listeners](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener)
-that listen for `click` events on the toggle buttons. Because the `nativeTheme`
-module only exposed in the Main process, you need to set up each listener's
-callback to use IPC to send messages to and handle responses from the Main
-process:
+And the `style.css` file:
+
+```css title='style.css'
+@media (prefers-color-scheme: dark) {
+  body { background: #333; color: white; }
+}
+
+@media (prefers-color-scheme: light) {
+  body { background: #ddd; color: black; }
+}
+```
+
+The example renders an HTML page with a couple elements. The `<strong id="theme-source">`
+ element shows which theme is currently selected, and the two `<button>` elements are the
+ controls. The CSS file uses the [`prefers-color-scheme`][prefers-color-scheme] media query
+ to set the `<body>` element background and text colors.
+
+Run the example using Electron Fiddle and then click the "Toggle Dark Mode" button; the app
+ should start alternating between a light and dark background color.
 
-* when the "Toggle Dark Mode" button is clicked, we send the
-`dark-mode:toggle` message (event) to tell the Main process to trigger a theme
-change, and update the "Current Theme Source" label in the UI based on the
-response from the Main process.
-* when the "Reset to System Theme" button is clicked, we send the
-`dark-mode:system` message (event) to tell the Main process to use the system
-color scheme, and update the "Current Theme Source" label to `System`.
+The `preload.js` script adds a new API to the `window` object called `darkMode`. This API
+ only exposes the necessary IPC channels to the renderer process.
+
+```js title='preload.js'
+const { contextBridge, ipcRenderer } = require('electron')
+
+contextBridge.exposeInMainWorld('darkMode', {
+  toggle: () => ipcRenderer.invoke('dark-mode:toggle'),
+  system: () => ipcRenderer.invoke('dark-mode:system')
+})
+```
 
-To add listeners and handlers, add the following lines to the `renderer.js` file:
+Through the `contextBridge`, the `preload.js` script exposes two IPC channels, `'dark-mode:toggle'`
+ and `'dark-mode:system'`, and assigns two methods, `toggle` and `system`, to the `darkMode` API.
+ Now the renderer process can communicate with the main process securely and perform the necessary
+ mutations to the `nativeTheme` object. For more information on IPC channels visit the
+ [context isolation](./context-isolation.md) documentation.
```

Comment:
I don't think the context isolation doc is the right one to use as a reference for IPC channels. 🤔 

We could probably just omit this entirely.
```suggestion
 mutations to the `nativeTheme` object.
```